packages/tldraw/src/lib/ui/components/InputModeMenu.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEditor, useValue } from '@tldraw/editor'
+import { useUiEvents } from '../context/events'
+import { TldrawUiMenuCheckboxItem } from './primitives/menus/TldrawUiMenuCheckboxItem'
+import { TldrawUiMenuGroup } from './primitives/menus/TldrawUiMenuGroup'
+import { TldrawUiMenuSubmenu } from './primitives/menus/TldrawUiMenuSubmenu'
+
+const MODES = ['auto', 'trackpad', 'mouse'] as const
+
+/** @public @react */
+export function InputModeMenu() {
+	const editor = useEditor()
+	const trackEvent = useUiEvents()
+
+	const inputMode = useValue('inputMode', () => editor.user.getUserPreferences().inputMode, [
+		editor,
+	])
+	const wheelBehavior = useValue('wheelBehavior', () => editor.getCameraOptions().wheelBehavior, [
+		editor,
+	])
+
+	const isModeChecked = (mode: string) => {
+		if (mode === 'auto') {
+			return inputMode === null
+		}
+		return inputMode === mode
+	}
+
+	const getLabel = (mode: string, wheelBehavior: 'zoom' | 'pan' | 'none') => {
```

Comment:
if `wheelBehavior` is set to `'none'` will we end up with a invalid label ID? if so we should add a special case in here

Reply:
yeah good point! I think the label without parenthesis works in this case? 

```ts
"action.toggle-auto-none": "Auto",
```

If we are happy with that, I'll add it and then merge :)